ntbs-integration-tests/ClinicalDetailsPageTests.cs

```diff
@@ -128,46 +155,128 @@ public async Task Post_ClearsConditionalInputValues_IfRadiosSetToOtherValue()
-            Assert.Equal(ValidationMessages.DateValidityRange(ValidDates.EarliestClinicalDate), result);
-        } 
+            Assert.Equal(errorMessage, result);
+        }
 
-        // TODO: Tests for full validation, all onget validate methods
+        [Theory]
+        [InlineData("true", ValidationMessages.DiseaseSiteIsRequired)]
+        [InlineData("false", "")]
+        public async Task ValidateNotificationSites_ReturnsExpectedResult(string shouldValidateFull, string validationResult)
+        {
+            // Arrange
+            var formData = new Dictionary<string, string>
+            {
+                ["shouldValidateFull"] = shouldValidateFull
+            };
+
+            // Act
+            var response = await client.GetAsync(BuildValidationPath(formData, "ValidateNotificationSites"));
+
+            // Assert
+            var result = (await response.Content.ReadAsStringAsync());
+            Assert.Equal(validationResult, result);
+        }
+
+        [Theory]
+        [InlineData("false", "123", ValidationMessages.StandardStringFormat)]
+        [InlineData("true", "", ValidationMessages.DiseaseSiteOtherIsRequired)]
+        [InlineData("false", "", "")]
+        public async Task ValidateNotificationSiteProperty_ReturnsExpectedResult(string shouldValidateFull, string value, string validationResult)
+        {
+            // Arrange
+            var formData = new Dictionary<string, string>
+            {
+                ["key"] = "SiteDescription",
+                ["value"] = value,
+                ["shouldValidateFull"] = shouldValidateFull
+            };
+
+            // Act
+            var response = await client.GetAsync(BuildValidationPath(formData, "ValidateNotificationSiteProperty"));
+
+            // Assert
+            var result = (await response.Content.ReadAsStringAsync());
+            Assert.Equal(validationResult, result);
+        }
+
+        [Fact]
+        public async Task ValidateClinicalDetailsYearComparison_ReturnsErrorIfNewYearEarlierThanExisting()
+        {
+             // Arrange
+            var existingYear = 1990; 
+            var formData = new Dictionary<string, string>
+            {
+                ["newYear"] = "1960",
+                ["existingYear"] = existingYear.ToString()
+            };
+
+            // Act
+            var response = await client.GetAsync(BuildValidationPath(formData, "ValidateClinicalDetailsYearComparison"));
+
+            // Assert
+            var result = (await response.Content.ReadAsStringAsync());
+            Assert.Equal(ValidationMessages.ValidYearLaterThanBirthYear(existingYear), result);
+        }
+
+        [Fact]
+        public async Task ValidateClinicalDetailsProperties_ReturnsErrorIfBothTreatmentsSetToTrue()
+        {
+             // Arrange
+            var keyValuePairs = "keyValuePairs[0][key]=IsShortCourseTreatment&keyValuePairs[0][value]=true&keyValuePairs[1][key]=IsMDRTreatment&keyValuePairs[1][value]=true";

```

Comment:
This is a bit of a nightmare to eyeball...
Not sure how much we can do without making it awefully complex, but perhaps something as simple as breaking it into a list:
```
new List () {
"keyValuePairs[0][key]=IsShortCourseTreatment",
"keyValuePairs[0][value]=true",
"keyValuePairs[1][key]=IsMDRTreatment",
"keyValuePairs[1][value]=true"
}.Join("&")
?

Reply:
Good shout, will do!

I did check whether there is a simple way to generate a query string from a c# object but there wasn't anything out-of-the box I could find so considering it's a test I thought it's okay to spell it out, but I definitely agree it is too unreadable right now